src/window.rs

```diff
@@ -1111,6 +1112,23 @@ impl Window {
         self.window.set_ime_purpose(purpose);
     }
 
+    /// Opens the IME input (soft keyboard) if the platform supports it.
+    /// Currently only supported on Android.
+    #[inline]
+    pub fn begin_ime_input(&self) {
+        self.window.begin_ime_input();
+    }
+
+    /// Hides the IME input (soft keyboard).
+    #[inline]
+    pub fn end_ime_input(&self) {
+        self.window.end_ime_input();
+    }
+
+    pub fn set_text_input_state(&self, state: TextInputState) {
+        self.window.set_text_input_state(state);
+    }
```

Comment:
Could you get a bit more details on what this particular function tries to do? In general, that's true that we'd nee a way to set an IME state, like position of the ime, surrounding text, etc.

I think what we probably should do is to have a `set_ime_state(state: Option<ImeState>)` where you can set that information + position of the input method, content hint, and similar.

Reply:
I've replaced this with a set_ime_surrounding_text function similar to the one from wayland you sent.

Reply:
Does android have a notion of atomic state changes? Because wayland has for example, and using some big `InputState` would be preferred for it, but I can refactor it myself later.

Reply:
I'm not sure what you mean, but I'd guess not. The android game text input library this relies on is quite simple. The idea is that this is called once when a text field is first focused and when the selection gets updated by the user.